src/helpers.js

```diff
@@ -36,13 +55,27 @@ function mergeHash(source, target, old, keepRemoved) {
 
     Object.keys(source).forEach(function (key) {
         if ( target[key] !== undefined ) {
-            if (typeof source[key] === 'object' && source[key].constructor !== Array) {
-                var nested = mergeHash( source[key], target[key], old[key], keepRemoved );
-                target[key] = nested.new;
-                old[key] = nested.old;
-            }
-            else {
-                target[key] = source[key];
+            if (typeof target[key] === 'object' && typeof target[key].msgstr !== 'undefined') {
```

Comment:
This is exactly what I was afraid of, it is becoming impossible to maintain. The code complexity is way to high, which will make changing it painful in the future. My head hurts just trying to keep track of what's going on here. I think this must be broken down, either as a separate helper or into smaller chunks with better SRP.